oauth/services/rhsso.py

```diff
@@ -74,3 +74,18 @@ def exchange_code_for_login(self, app_config, http_client, code, redirect_suffix
                 )
 
         return sub, lusername, email_address, additional_info
+
+    def get_user_id(self, decoded_id_token: dict) -> str:
+        """
+        RHSSO now uses digit values for `sub` field. Old value is retained in `deprecated_sub`.
+        Returns `deprecated_sub` if `sub` is a digit, otherwise returns `sub`.
+        """
+        sub = decoded_id_token.get("sub")
+        if not sub:
+            raise OAuthLoginException("Token missing 'sub' field")
+
+        if sub.isdigit():
+            # For older tokens, use deprecated_sub if available, fallback to sub
+            return decoded_id_token.get("deprecated_sub", sub)
+        else:
```

Comment:
nit: else is not needed

Reply:
just the `return` statement is enough

Reply:
we can say that if `deprecated_sub` exists, always return that. else return the `sub`

Reply:
ack, removed the `sub.isdigit()` check and instead use `deprecated_sub` if exists, else the `sub`